aws/resource_aws_imagebuilder_image_pipeline_test.go

```diff
@@ -0,0 +1,869 @@
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAwsImageBuilderImagePipelineConfigStatus(rName, imagebuilder.PipelineStatusEnabled),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsImageBuilderImagePipelineExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "status", imagebuilder.PipelineStatusEnabled),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAwsImageBuilderImagePipeline_Tags(t *testing.T) {
+	rName := acctest.RandomWithPrefix("tf-acc-test")
+	resourceName := "aws_imagebuilder_image_pipeline.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAwsImageBuilderImagePipelineDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsImageBuilderImagePipelineConfigTags1(rName, "key1", "value1"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsImageBuilderImagePipelineExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAwsImageBuilderImagePipelineConfigTags2(rName, "key1", "value1updated", "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsImageBuilderImagePipelineExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1updated"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+			{
+				Config: testAccAwsImageBuilderImagePipelineConfigTags1(rName, "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsImageBuilderImagePipelineExists(resourceName),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckAwsImageBuilderImagePipelineDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).imagebuilderconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_imagebuilder_image_pipeline" {
+			continue
+		}
+
+		input := &imagebuilder.GetImagePipelineInput{
+			ImagePipelineArn: aws.String(rs.Primary.ID),
+		}
+
+		output, err := conn.GetImagePipeline(input)
+
+		if tfawserr.ErrCodeEquals(err, imagebuilder.ErrCodeResourceNotFoundException) {
+			continue
+		}
+
+		if err != nil {
+			return fmt.Errorf("error getting Image Builder Image Pipeline (%s): %w", rs.Primary.ID, err)
+		}
+
+		if output != nil {
+			return fmt.Errorf("Image Builder Image Pipeline (%s) still exists", rs.Primary.ID)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckAwsImageBuilderImagePipelineExists(resourceName string) resource.TestCheckFunc {
```

Comment:
Should the image pipeline be returned using from the exists function?

Reply:
Created https://github.com/hashicorp/terraform-provider-aws/issues/16435 and added to technical research board.

Reply:
To provide more color here, I think we can move away from those "CheckExists" functions altogether since import testing accomplishes the same thing, rather than worrying about creating those or making them consistent.